File: Searching/Linear_Search.cpp
// TIME COMPLEXITY : O(n)

#include <iostream>
using namespace std;

bool search(int arr[], int size, int key)
{
    for (int i = 0; i < size; i++)
    {
        if (arr[i] == key)
            return 1;
    }

    return 0;
}

int main()
{
    int arr[10] = {5, 2, 3, 8, 9, 5, 6, 4, 2, 55};
    bool found = search(arr, 10, 55);

    if (found)
        cout << "Key is present";
    else
        cout << "Key is not present";

    return 0;
}

File: Searching/LinearSearch.js
/**
 * Implements Linear Search algorithm
 * @param {number[]} arr - Array to search in
 * @param {number} target - Element to find
 * @returns {number} - Index of target element or -1 if not found
 */
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;
        }
    }
    return -1;
}

// Example usage
const arr = [5, 2, 9, 1, 7, 6, 3];
const target = 7;
console.log(`Index of ${target} in array:`, linearSearch(arr, target)); 